src/main/java/edu/cornell/kfs/tax/dataaccess/impl/TransactionRowDvBuilder.java

```diff
@@ -433,49 +433,53 @@ void updateTransactionRowsFromWorkflowDocuments(ResultSet rs, T summary) throws
                 
                 if (processCurrentRow) {
                     // Finish initialization.
-                    vendorTaxNumber = rs.getString(detailRow.vendorTaxNumber.index);
-                    checkStubText = rs.getString(detailRow.dvCheckStubText.index);
+                    SecondPassAttributeUpdateValues updatedAttributeValues = new SecondPassAttributeUpdateValues();
+                    vendorTaxNumber = rs.getString(rawDetailRow.vendorTaxNumber.index);
+                    checkStubText = rs.getString(rawDetailRow.dvCheckStubText.index);
                     checkStubMatcher = nonPrintableCharsPattern.matcher((checkStubText != null) ? checkStubText : KFSConstants.EMPTY_STRING);
                     
                     // Check for null objects as needed, and get the initiator's principal name.
                     initiatorPrincipalName = checkForEntityAndAccountAndOrgExistence(initiatorPrincipalId,
-                            rs.getString(detailRow.chartCode.index), rs.getString(detailRow.accountNumber.index), summary);
+                            rs.getString(rawDetailRow.chartCode.index), rs.getString(rawDetailRow.accountNumber.index), summary);
                     
                     // If vendor tax number is blank, then replace with a generated value accordingly.
                     if (StringUtils.isBlank(vendorTaxNumber)) {
-                        vendorTaxNumber = getReplacementVendorTaxNumber(rs.getString(detailRow.payeeId.index), summary);
-                        rs.updateString(detailRow.vendorTaxNumber.index, vendorTaxNumber);
+                        vendorTaxNumber = getReplacementVendorTaxNumber(rs.getString(rawDetailRow.payeeId.index), summary);
+                        updatedAttributeValues.putStringAttributeForUpdating(rawDetailRow.vendorTaxNumber.index, vendorTaxNumber);

```

Comment:
I tried to keep the processing as close to the original as possible with these changes.
Could you please take a look at this area again while considering the following:

- The first pass table is now the `RawTransactionDetailRow` and the final table is the second pass table is the `TransactionDetailRow`.
- While setting up the second pass data (that **does not yet exist** in any second pass data structure or table), we are looping over and processing information from the first pass select statement that has a valid ResultSet defined for a  `RawTransactionDetailRow`. While in that loop, we are trying to determine whether to modify that first pass data before inserting it into the second pass table but there is nothing defined yet for the `TransactionDetailRow` so we do not have any column indexes for that second pass table , thus we do not know which array element of the insert statement each data value should be placed in. 
- But, the `TransactionDetailRow` table and `RawTransactionDetailRow` table are defined to be identical so the array element (or column positions) for both tables would be the same, meaning that the values from the ResultSet we do have access to (and is defined) would match column for column to be able to correctly setup the insert statement for the second table.

The reason I am asking for a second set of eyes here again is because it took a while for me to get this area to work at all.

Reply:
I was under the impression that we already knew in advance what the `TransactionDetailRow` SQL statement column indexes would be, given that your `TransactionRowBuilder` changes are causing the second-pass data to default to the values from the first-pass data (excluding the primary key value). Your new column-updating helper object only seems to be tracking the values that should override the defaults described earlier, but since it's meant for configuring the `TransactionDetailRow` insertion, it should reference the `TransactionDetailRow` indexes when its elements get populated. I realize that the first-pass and second-pass objects are identical for the moment; however, once we implement the Payment Method Code addition in a future user story, the two objects will start to diverge a bit.

I would recommend changing the `putXXXAttributeForUpdating()` calls to populate the second-pass indexes instead. If you have a strong opinion about keeping them as-is for now, then we'll need to implement the change in the follow-up Payment Method Code user story.